src/main/java/com/mraof/minestuck/data/worldgen/MSPlacedFeatureProvider.java

```diff
@@ -0,0 +1,305 @@
+				worldGenModifiers(RarityFilter.onAverageOnceEvery(56), PlacementUtils.HEIGHTMAP_TOP_SOLID)));
+		consumer.accept(COBBLESTONE_SURFACE_DISK, placed(features, MSCFeatures.COBBLESTONE_SURFACE_DISK,
+				worldGenModifiers(RarityFilter.onAverageOnceEvery(20), PlacementUtils.HEIGHTMAP_TOP_SOLID)));
+		consumer.accept(STONE_SURFACE_DISK, placed(features, MSCFeatures.STONE_SURFACE_DISK,
+				worldGenModifiers(RarityFilter.onAverageOnceEvery(20), PlacementUtils.HEIGHTMAP_TOP_SOLID)));
+		consumer.accept(END_GRASS_SURFACE_DISK, placed(features, MSCFeatures.END_GRASS_SURFACE_DISK,
+				worldGenModifiers(RarityFilter.onAverageOnceEvery(5), PlacementUtils.HEIGHTMAP_TOP_SOLID)));
+		consumer.accept(END_STONE_SURFACE_DISK, placed(features, MSCFeatures.END_STONE_SURFACE_DISK,
+				worldGenModifiers(RarityFilter.onAverageOnceEvery(4), PlacementUtils.HEIGHTMAP_TOP_SOLID)));
+		
+		//these are similar to the ones in CavePlacements, but with edits to the height at which they generate and count
+		consumer.accept(DRIPSTONE_CLUSTER, placed(features, CaveFeatures.DRIPSTONE_CLUSTER,
+				worldGenModifiers(CountPlacement.of(UniformInt.of(22, 74)), HeightRangePlacement.uniform(VerticalAnchor.bottom(), VerticalAnchor.absolute(60)))));
+		consumer.accept(OCEANIC_DRIPSTONE_CLUSTER, placed(features, CaveFeatures.DRIPSTONE_CLUSTER,
+				worldGenModifiers(CountPlacement.of(UniformInt.of(42, 96)), HeightRangePlacement.uniform(VerticalAnchor.bottom(), VerticalAnchor.absolute(48)))));
+		consumer.accept(LARGE_DRIPSTONE, placed(features, CaveFeatures.LARGE_DRIPSTONE,
+				worldGenModifiers(CountPlacement.of(UniformInt.of(10, 48)), HeightRangePlacement.uniform(VerticalAnchor.bottom(), VerticalAnchor.absolute(64)))));
+		consumer.accept(POINTED_DRIPSTONE, placed(features, CaveFeatures.POINTED_DRIPSTONE,
+				worldGenModifiers(CountPlacement.of(UniformInt.of(100, 200)), HeightRangePlacement.uniform(VerticalAnchor.bottom(), VerticalAnchor.absolute(60)))));
+		consumer.accept(OCEANIC_POINTED_DRIPSTONE, placed(features, CaveFeatures.POINTED_DRIPSTONE,
+				worldGenModifiers(CountPlacement.of(UniformInt.of(164, 256)), HeightRangePlacement.uniform(VerticalAnchor.bottom(), VerticalAnchor.absolute(48)))));
+		consumer.accept(LUSH_CAVES_VEGETATION, placed(features, CaveFeatures.MOSS_PATCH,
+				worldGenModifiers(CountPlacement.of(125), HeightRangePlacement.uniform(VerticalAnchor.bottom(), VerticalAnchor.absolute(64)), EnvironmentScanPlacement.scanningFor(Direction.DOWN, BlockPredicate.solid(), BlockPredicate.ONLY_IN_AIR_PREDICATE, 12), RandomOffsetPlacement.vertical(ConstantInt.of(1)))));
+		consumer.accept(LUSH_CAVES_CEILING_VEGETATION, placed(features, CaveFeatures.MOSS_PATCH_CEILING,
+				worldGenModifiers(CountPlacement.of(125), HeightRangePlacement.uniform(VerticalAnchor.bottom(), VerticalAnchor.absolute(64)), EnvironmentScanPlacement.scanningFor(Direction.UP, BlockPredicate.solid(), BlockPredicate.ONLY_IN_AIR_PREDICATE, 12), RandomOffsetPlacement.vertical(ConstantInt.of(-1)))));
```

Comment:
Still trying to determine if it was already like that or if there was a mistranslation, but this generates at a visually higher rate than intended

Reply:
It looks to be the same as the [original](https://github.com/mraof/Minestuck/blob/948469d4e40072ef9e85476fa113069946e6803a/src/main/java/com/mraof/minestuck/world/gen/feature/MSPlacedFeatures.java#L116-L117), so I assume that it was already like that.